frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts

```diff
@@ -10,18 +13,34 @@ import { questionIdToTradPath } from "../utils";
 export const useOnboardingQuestion = (questionId: QuestionId) => {
   const { updateOnboarding } = useOnboarding();
   const { t } = useTranslation("chat");
+  const [isAnswerRequested, setIsAnswerRequested] = useState(false);
 
   const onboardingStep = questionIdToTradPath[questionId];
-
   const question = t(`onboarding.${onboardingStep}`);
-
-  const { addQuestion } = useChat();
+  const { updateStreamingHistory } = useChatContext();
+
+  const { lastStream } = useStreamText({
+    text: t(`onboarding.answer.${onboardingStep}`),
+    enabled: isAnswerRequested,
+  });
+
+  useEffect(() => {
+    if (isAnswerRequested) {
+      const chatMessage: ChatMessage = {
+        chat_id: questionId,
+        message_id: questionId,
+        user_message: question,
+        assistant: lastStream,
+        message_time: Date.now().toLocaleString(),
+        brain_name: "Quivr",
+      };
+      void updateStreamingHistory(chatMessage);
+    }
+  }, [isAnswerRequested, question, questionId, lastStream]);
 
   const handleSuggestionClick = async () => {
-    await Promise.all([
-      addQuestion(question),
-      updateOnboarding({ [questionId]: false }),
-    ]);
+    setIsAnswerRequested(true);
```

Comment:
Here replace "addQuestion(question)" with either create new chat + update chat or with update chat only if the question is asked in an already existing chat